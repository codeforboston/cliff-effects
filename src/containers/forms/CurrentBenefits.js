import { connect } from 'react-redux';

import { CurrentBenefitsStep } from '../../forms/CurrentBenefits';
import { setHasBenefit } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),

    USState: state.getIn([
      'geography',
      'state',
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setHasBenefit({ benefit, value }) {
      dispatch(
        setHasBenefit({
          time: 'current',
          benefit,
          value,
        })
      );
    },
  };
};

const CurrentBenefitsContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentBenefitsStep);

export { CurrentBenefitsContainer as CurrentBenefitsStep };
