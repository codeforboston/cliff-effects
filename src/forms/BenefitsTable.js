// REACT COMPONENTS
import React from 'react';
import {
  Table
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/federal/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var currentClient = cloneDeep( props.client ),
      futureClient  = cloneDeep( currentClient );
  // Hack because everything's using future value instead
  // of just using current value.
  currentClient.future.earned = currentClient.current.earned;




  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( futureClient ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( futureClient ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( currentClient.current.earned * 12 ),
      incomeFuture        = Math.round( futureClient.future.earned * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;


const   columnHeaderStyle = {
                            background: 'rgba(0, 181, 173, 1)',
                            color: 'white',
                            fontSize: '1.3em',
                            fontWeight: 900,
                            textAlign: 'center',
                            borderRadius: 'inherit',
                            letterSpacing: '0.02em',
                            }
      , totalsRowStyle    = {
                            borderTop: "2px solid rgba(0, 181, 173, 1)",
                            fontWeight: 700,
                            fontSize: '1.1em',
                            padingTop: '0.25em'
                              }
      , rowHeaderStyle    = {
                    fontSize: '1.1em',
                    fontWeight: 700,
                    textAlign: 'left'
                    }
      , totalsRowHeaderStyle = {
                    fontSize: '1.2em',
                    fontWeight: 700,
                    textAlign: 'left',
                    borderTop: "2px solid rgba(0, 181, 173, 1)",
                    padingTop: '0.25em'


      };






const SNAPBenefitRow = function( props ){
  if(!currentClient.current.hasSnap ) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>SNAP</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };

const Sec8BenefitRow  = function( props ){  
  if(!currentClient.current.hasHousing) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Section 8 Housing</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
      </Table.Row>
    ) 
};

const TotalBenefitsRow = function( props ){
  if(!currentClient.current.hasSnap || !currentClient.current.hasHousing) return (null)
    return(
      <Table.Row>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Total Benefits</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const IncomeRow = function ( props ) { 
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Income</Table.Cell>
        <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const TotalsRow = function ( props ) {
    return (
      <Table.Row style={{border: 'none'}}>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Net Total</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };


  return (
    <wrapper>
      <Table celled>
       <Table.Header>
          <Table.Row >
            <Table.Cell style={columnHeaderStyle} width={3}>Benefit</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Current Benefits</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>New Estimate</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Difference</Table.Cell>
          </Table.Row>
      </Table.Header>
        <Table.Body>
          <SNAPBenefitRow client={currentClient} />
          <Sec8BenefitRow client={currentClient} />
          <TotalBenefitsRow client={currentClient} />
          <IncomeRow />
          <TotalsRow />
        </Table.Body>
      </Table>
    </wrapper>
  )

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
