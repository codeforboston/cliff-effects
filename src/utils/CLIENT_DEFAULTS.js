import { cloneDeep } from 'lodash';

/**
 * Default client values.
 * MUST ALWAYS BE VALID
 * 
 * Note: A 'positive number' money value can have more than
 *     two decimal places.
 */
const CLIENT_DEFAULTS = {

  // Property Name                | Default Value   | Valid Value Description

  current: {
    // Current programs
    hasSnap:                        false,            // boolean
    hasHousing:                     false,            // boolean
    // Household
    // 'm_' for 'member'
    household:  [{                  // 'head' member  // one or more member objects
        m_age:                      30,               // whole number
        m_role:                     'head',           // 'head', 'spouse', 'member'
        m_disabled:                 false             // boolean
    }],
    // Income
    earned:                         0,                // positive number (can have more than two decimal places)
    TAFDC:                          0,                // positive number
    SSI:                            0,                // positive number
    SSDI:                           0,                // positive number
    childSupportIn:                 0,                // positive number
    unemployment:                   0,                // positive number
    workersComp:                    0,                // positive number
    pension:                        0,                // positive number
    socialSecurity:                 0,                // positive number
    alimony:                        0,                // positive number
    otherIncome:                    0,                // positive number
    incomeExclusions:               0,                // positive number
    // Expenses
    childDirectCare:                0,                // positive number
    childBeforeAndAfterSchoolCare:  0,                // positive number
    childTransportation:            0,                // positive number
    childOtherCare:                 0,                // positive number
    earnedBecauseOfChildCare:       0,                // positive number
    childSupportPaidOut:            0,                // positive number
    adultDirectCare:                0,                // positive number
    adultTransportation:            0,                // positive number
    adultOtherCare:                 0,                // positive number
    disabledAssistance:             0,                // positive number
    earnedBecauseOfAdultCare:       0,                // positive number
    disabledMedical:                0,                // positive number
    otherMedical:                   0,                // positive number
    /** @todo When client has section 8, switch this to 'housingVoucher' */
    /** @todo Make shelter expense values more robust. */
    shelter:                        'homeless',       // 'homeless', 'homeowner', 'renter', 'housingVoucher'
    contractRent:                   0,                // positive number
    rentShare:                      0,                // positive number
    rent:                           0,                // positive number
    mortgage:                       0,                // positive number
    housingInsurance:               0,                // positive number
    propertyTax:                    0,                // positive number
    climateControl:                 false,            // boolean
    nonHeatElectricity:             false,            // boolean
    phone:                          false,            // boolean
    fuelAssistance:                 false,            // boolean  Note: Input is 'yes', 'no', but should be being converted
    otherExpenses:                  0                 // positive number
  },

  // Note: A `future` value will default to the client's `current` value until a user changes it explicitly
  future: {},                       // same structure // same

};

CLIENT_DEFAULTS.future = cloneDeep( CLIENT_DEFAULTS.current );


export { CLIENT_DEFAULTS };
