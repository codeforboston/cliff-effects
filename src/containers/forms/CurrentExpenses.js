import { connect } from 'react-redux';

import { CurrentExpensesStep } from '../../forms/CurrentExpenses';

import { setCashValue, setHousingType, setPaysUtility, setGetsFuelAssistance } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setExpenseValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },

    setHousingType({ housingType }) {
      dispatch(
        setHousingType({
          time: 'current',
          housingType,
        })
      );
    },

    setPaysUtility({ utility, paysUtility }) {
      dispatch(
        setPaysUtility({
          time: 'current',
          utility,
          paysUtility,
        })
      );
    },

    setGetsFuelAssistance({ getsAssistance }) {
      dispatch(
        setGetsFuelAssistance({
          time: 'current',
          getsAssistance,
        })
      );
    },
  };
};

const CurrentExpensesStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentExpensesStep);

export { CurrentExpensesStepContainer as CurrentExpensesStep };
