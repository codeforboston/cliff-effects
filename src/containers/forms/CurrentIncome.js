import { connect } from 'react-redux';

import { CurrentIncomeStep } from '../../forms/CurrentIncome';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setIncomeValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },
  };
};

const CurrentIncomeStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentIncomeStep);

export { CurrentIncomeStepContainer as CurrentIncomeStep };
