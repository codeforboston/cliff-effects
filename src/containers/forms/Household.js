import { connect } from 'react-redux';

import { HouseholdStep } from '../../forms/Household';
import {
  addMember,
  removeMember,
  setMemberIsDisabled,
  setMemberRole,
  setMemberAge,
} from '../../actions';

const time = 'current';

const mapStateToProps = (state) => {
  return {
    household: state.getIn([
      'client',
      'current',
      'household', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMemberAge({ index, age }) {
      dispatch(
        setMemberAge({
          time,
          index,
          age,
        })
      );
    },

    setMemberIsDisabled({ index, isDisabled }) {
      dispatch(
        setMemberIsDisabled({
          time,
          index,
          isDisabled,
        })
      );
    },

    setMemberRole({ index, role }) {
      dispatch(
        setMemberRole({
          time,
          index,
          role,
        })
      );
    },

    removeMember({ index }) {
      dispatch(
        removeMember({
          time,
          index,
        })
      );
    },

    addMember({ member }) {
      dispatch(
        addMember({
          time,
          member,
        })
      );
    },
  };
};

const HouseholdStepContainer = connect(mapStateToProps, mapDispatchToProps)(HouseholdStep);

export { HouseholdStepContainer as HouseholdStep };
