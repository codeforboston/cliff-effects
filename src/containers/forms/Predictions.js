import { connect } from 'react-redux';

import { PredictionsStep } from '../../forms/Predictions';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return { client: state.get('client') };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setPredictionValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'future',
          name,
          value,
        })
      );
    },
  };
};

const PredictionsStepContainer = connect(mapStateToProps, mapDispatchToProps)(PredictionsStep);

export { PredictionsStepContainer as PredictionsStep };
