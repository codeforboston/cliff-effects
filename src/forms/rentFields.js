import React, { Component } from 'react';
import { MonthlyCashFlowRow } from './formHelpers';

import { isPositiveNumber } from '../utils/validators';


class RentShareField extends Component {
  state = { valid: true,
    message: null };

  validation = (ownValue) => {
    var message = null, valid = true;

    let isPosNum = isPositiveNumber(ownValue);
    if (!isPosNum) { valid = false; }
    else {
      valid = ownValue <= this.props.timeState[ 'contractRent' ];
      if (!valid) { message = 'Rent share must be less than contract rent'; }
    }

    this.setState({ valid: valid,
      message: message });
    return valid;
  };

  onBlur = (evnt) => {
    this.setState({ valid: true,
      message: null });
  };

  render() {
    const { timeState, setClientProperty } = this.props,
      { valid, message } = this.state;

    const inputProps = {
        name:       'rentShare',
        validation: this.validation,
        onBlur:     this.onBlur,
      },
      rowProps = {
        label:    'Your Monthly Rent Share (how much of the total rent you have to pay)',
        validRow: valid,
        message:  message,
      };

    return (
      <MonthlyCashFlowRow
        inputProps        = { inputProps }
        baseValue         = { timeState[ 'rentShare' ] }
        includes          = {[
          'monthly', 
        ]}
        setClientProperty = { setClientProperty }
        rowProps          = { rowProps } />
    );
  }
}  // End <RentShareField>


class ContractRentField extends Component {
  state = { valid: true,
    message: null };

  validation = (ownValue) => {
    var message = null, valid = true;

    let isPosNum = isPositiveNumber(ownValue);
    if (!isPosNum) { valid = false; }
    else {
      valid = ownValue >= this.props.timeState[ 'rentShare' ];
      if (!valid) { message = 'Rent share must be less than contract rent'; }
    }

    this.setState({ valid: valid,
      message: message });
    return valid;
  };

  onBlur = (evnt) => {
    this.setState({ valid: true,
      message: null });
  };

  render() {
    const { timeState, setClientProperty } = this.props,
      { valid, message } = this.state;

    const inputProps = {
        name:       'contractRent',
        validation: this.validation,
        onBlur:     this.onBlur,
      },
      rowProps = {
        label:    'Monthly Contract Rent (the total rent for your apartment)',
        validRow: valid,
        message:  message,
      };

    return (
      <MonthlyCashFlowRow
        inputProps        = { inputProps }
        baseValue         = { timeState[ 'contractRent' ] }
        includes          = {[
          'monthly', 
        ]}
        setClientProperty = { setClientProperty }
        rowProps          = { rowProps } />
    );
  }
}  // End <ContractRentField>


const PlainRentRow = function ({ timeState, setClientProperty }) {

  const inputProps = {
      name:       'rent',
      validation: isPositiveNumber,
      onBlur:     function () {},
    },
    rowProps = {
      label:    'Monthly Rent',
      validRow: true,
      message:  null,
    };

  return (
    <MonthlyCashFlowRow
      inputProps        = { inputProps }
      baseValue         = { timeState[ 'rent' ] }
      includes          = {[
        'monthly', 
      ]}
      setClientProperty = { setClientProperty }
      rowProps          = { rowProps } />
  );

};  // End <PlainRentRow>


export {
  ContractRentField,
  RentShareField,
  PlainRentRow,
};
