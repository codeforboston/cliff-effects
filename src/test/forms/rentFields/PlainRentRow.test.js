import React from 'react';
import { mount } from 'enzyme';
import { cloneDeep } from 'lodash';

import { CLIENT_DEFAULTS } from '../../../utils/CLIENT_DEFAULTS';
import { PlainRentRow } from '../../../forms/rentFields';

describe('<PlainRentRow>', () => {
  it('renders', () => {
    const props = {
      timeState:         cloneDeep(CLIENT_DEFAULTS.current),
      updateClientValue: jest.fn(),
    };
    expect(() => {
      mount(<PlainRentRow { ...props } />);
    }).not.toThrow();
  });
});
