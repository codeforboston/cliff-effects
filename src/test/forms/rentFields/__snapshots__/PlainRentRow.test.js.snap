// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainRentRow> matches snapshot 1`] = `
<PlainRentRow
  timeState={
    Object {
      "SSDI": 0,
      "SSI": 0,
      "TAFDC": 0,
      "adultDirectCare": 0,
      "adultOtherCare": 0,
      "adultTransportation": 0,
      "alimony": 0,
      "childBeforeAndAfterSchoolCare": 0,
      "childDirectCare": 0,
      "childOtherCare": 0,
      "childSupportIn": 0,
      "childSupportPaidOut": 0,
      "childTransportation": 0,
      "climateControl": false,
      "contractRent": 0,
      "disabledAssistance": 0,
      "disabledMedical": 0,
      "earned": 0,
      "earnedBecauseOfAdultCare": 0,
      "earnedBecauseOfChildCare": 0,
      "fuelAssistance": false,
      "hasSection8": false,
      "hasSnap": false,
      "household": Array [
        Object {
          "m_age": 30,
          "m_disabled": false,
          "m_role": "head",
        },
      ],
      "housing": "homeless",
      "housingInsurance": 0,
      "mortgage": 0,
      "nonHeatElectricity": false,
      "otherExpensesCable": 0,
      "otherExpensesCareProducts": 0,
      "otherExpensesClothes": 0,
      "otherExpensesEntertainment": 0,
      "otherExpensesFood": 0,
      "otherExpensesMedical": 0,
      "otherExpensesOther": 0,
      "otherExpensesPhone": 0,
      "otherExpensesTransport": 0,
      "otherExpensesUtilities": 0,
      "otherIncome": 0,
      "otherMedical": 0,
      "pension": 0,
      "phone": false,
      "propertyTax": 0,
      "rent": 0,
      "rentShare": 0,
      "socialSecurity": 0,
      "unemployment": 0,
      "wantsToSeeOtherExpenses": false,
      "workersComp": 0,
    }
  }
  updateClientValue={[Function]}
>
  <MonthlyCashFlowRow
    baseValue={0}
    includes={
      Array [
        "monthly",
      ]
    }
    inputProps={
      Object {
        "displayValidator": [Function],
        "name": "rent",
        "onBlur": [Function],
        "storeValidator": [Function],
      }
    }
    rowProps={
      Object {
        "label": "Monthly Rent",
        "message": null,
        "validRow": true,
      }
    }
    updateClientValue={[Function]}
  >
    <CashFlowContainer
      label="Monthly Rent"
      message={null}
      validRow={true}
    >
      <FormField
        className="cashflow"
        inline={true}
      >
        <div
          className="inline field cashflow"
        >
          <ManagedNumberField
            className="cashflow-column"
            displayValidator={[Function]}
            format={[Function]}
            name="rent"
            onBlur={[Function]}
            otherData={
              Object {
                "interval": "monthly",
              }
            }
            store={[Function]}
            storeValidator={[Function]}
            value={0}
          >
            <FormInput
              as={[Function]}
              className="cashflow-column output-number"
              control={[Function]}
              error={false}
              name="rent"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              value="0.00"
            >
              <FormField
                className="cashflow-column output-number"
                control={[Function]}
                error={false}
                name="rent"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value="0.00"
              >
                <div
                  className="field cashflow-column output-number"
                >
                  <Input
                    name="rent"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value="0.00"
                  >
                    <div
                      className="ui input"
                    >
                      <input
                        key="text"
                        name="rent"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="text"
                        value="0.00"
                      />
                    </div>
                  </Input>
                </div>
              </FormField>
            </FormInput>
          </ManagedNumberField>
          <div
            className="cashflow-column cashflow-column-last-child"
          >
            <label>
              Monthly Rent
            </label>
          </div>
          <InvalidMessage
            message={null}
            validRow={true}
          />
        </div>
      </FormField>
    </CashFlowContainer>
  </MonthlyCashFlowRow>
</PlainRentRow>
`;
